server/storage/schema/schema_test.go

```diff
@@ -192,12 +192,10 @@ func TestMigrate(t *testing.T) {
 			expectErrorMsg: "cannot downgrade storage, WAL contains newer entries",
 		},
 		{
-			name:           "Downgrading v3.5 to v3.4 is not supported as schema was introduced in v3.6",
-			version:        version.V3_5,
-			targetVersion:  version.V3_4,
-			expectVersion:  nil,
-			expectError:    true,
-			expectErrorMsg: `cannot create migration plan: version "3.5.0" is not supported`,
+			name:          "Downgrading v3.5 to v3.4 is supported",
+			version:       version.V3_5,
+			targetVersion: version.V3_4,
+			expectVersion: nil,
```

Comment:
Hmm this test was not written with assumption storage version is empty. Maybe we could separate validation of storage version field (one read from UnsafeReadStorageVersion) and storage version it's compatible with (one inferred by fields present like `UnsafeDetectSchemaVersion`) 

Can you rename `expectVersion` to `expectedStorageVersion` and comment that v3.4 didn't have storage version. And update `UnsafeDetectSchemaVersion` to handle v3.5 use it to test for compatibility?